### PR TITLE
Zhengrui xia

### PR DESCRIPTION
**Pull request by lemadmax**

### CODE REVIEW

**Comment by kdokm** [comment]
Sorry but I don't think some of your changes to 'get' and 'list' are right.
For 'get', if the user id is invalid, I think it will raise 404 when we call find_shopcart(), before you perform check for 400 BAD_REQUEST. As you can see, it returns 404 in the corresponding test case.
For 'list', all_shopcart() is the one that finds all shopcart id with items, while all() finds all item records for shopcarts. I don't see why you want to change all_shopcart() to all(). And the original test case is correct since we have setup() method that will reset the table before each test.

**Review by kdokm — CHANGES_REQUESTED** [review]
As said in the previous comment

**Comment by kdokm** [comment]
> Sorry but I don't think some of your changes to 'get' and 'list' are right.
> For 'get', if the user id is invalid, I think it will raise 404 because it doesn’t fit the url specification, and will not perform check for 400 BAD_REQUEST. As you can see, it returns 404 in the corresponding test case.
> For 'list', all_shopcart() is the one that finds all shopcart id with items, while all() finds all item records for shopcarts. I don't see why you want to change all_shopcart() to all(). And the original test case is correct since we have setup() method that will reset the table before each test.

Maybe ‘get’ is ok if we delete the int limitation of user_id in the url, as I noticed #80 has put the check inside the method and return 400. Although I think we can wait for the prof’s reply  in homework channel.

File: service/utils/error_handlers.py [hunk 1]
```diff
@@ -29,13 +51,11 @@ def request_validation_error(error):
 
 @app.errorhandler(status.HTTP_400_BAD_REQUEST)
 def bad_request(error):
-    """Handles bad requests with 400_BAD_REQUEST"""
+    """Handles bad reuests with 400_BAD_REQUEST"""
```
Comment (TimothyXu) [inline service/utils/error_handlers.py:54]:
```suggestion
    """Handles bad requests with 400_BAD_REQUEST"""
```

File: service/utils/error_handlers.py [hunk 2]
```diff
@@ -44,19 +64,19 @@ def bad_request(error):
...
+    return make_response(
         jsonify(status=status.HTTP_404_NOT_FOUND, error="Not Found", message=message),
         status.HTTP_404_NOT_FOUND,
     )
 
 
 @app.errorhandler(status.HTTP_405_METHOD_NOT_ALLOWED)
 def method_not_supported(error):
-    """Handles unsupported HTTP methods with 405_METHOD_NOT_SUPPORTED"""
+    """Handles unsuppoted HTTP methods with 405_METHOD_NOT_SUPPORTED"""
```
Comment (TimothyXu) [inline service/utils/error_handlers.py:76]:
```suggestion
    """Handles unsupported HTTP methods with 405_METHOD_NOT_SUPPORTED"""
```

File: service/utils/error_handlers.py [hunk 3]
```diff
@@ -68,10 +88,10 @@ def method_not_supported(error):
 
 @app.errorhandler(status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
 def mediatype_not_supported(error):
-    """Handles unsupported media requests with 415_UNSUPPORTED_MEDIA_TYPE"""
+    """Handles unsuppoted media requests with 415_UNSUPPORTED_MEDIA_TYPE"""
```
Comment (TimothyXu) [inline service/utils/error_handlers.py:91]:
```suggestion
    """Handles unsupported media requests with 415_UNSUPPORTED_MEDIA_TYPE"""
```

**Comment by TimothyXu** [comment]
Apart from a few typos everything looks good, but the only thing keeping me from merging it is that the test coverage is below 95% - there are quite a few lines in log_handlers.py that have no tests and according to homework instructions we cannot accept any pull requests without 95% code coverage :(

**Comment by lemadmax** [comment]
> Apart from a few typos everything looks good, but the only thing keeping me from merging it is that the test coverage is below 95% - there are quite a few lines in log_handlers.py that have no tests and according to homework instructions we cannot accept any pull requests without 95% code coverage :( Also you should merge new changes on main branch. I'll try to patch up the missing test for routes.py first, and the missing tests for 3 lines in models.py, then that'll be enough to get us to 95%

Above 95% now

**Pull request merged**